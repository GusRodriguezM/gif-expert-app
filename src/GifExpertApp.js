import React, {useState} from 'react';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch Man', 'Samurai X', 'Dragon Ball']);

    const handleAdd = () => {
        setCategories(categories => [...categories, 'Demon Slayer']);
    }

    return (
        <>
            <h2>Gif Expert App</h2>
            <hr/>

            <button onClick={handleAdd}>Agregar</button>

            <ol>
                {
                    categories.map(category => {
                        return <li key={category}>{category}</li>
                    })
                }
            </ol>
        </>
    )
}

export default GifExpertApp;