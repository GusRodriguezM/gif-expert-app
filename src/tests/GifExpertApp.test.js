import React from 'react';
import { shallow } from "enzyme";
import '@testing-library/jest-dom'
import GifExpertApp from "../GifExpertApp";

describe('Tests in <GifExpertApp/> component', () => {
   
    test('should render <GifExpertApp/> component correctly', () => {
       
        const wrapper = shallow(<GifExpertApp/>);
        
        expect(wrapper).toMatchSnapshot();
        
    });

    test('should render a list of categories', () => {
       
        const categories = ['One Punch Man', 'Dragon Ball'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    });
    
});