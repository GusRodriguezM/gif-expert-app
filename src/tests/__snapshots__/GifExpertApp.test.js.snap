// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp/> component should render <GifExpertApp/> component correctly 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests in <GifExpertApp/> component should render a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch Man"
      key="One Punch Man"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
